keps/1788-pass-parameters-to-ProvReq/README.md

```diff
@@ -0,0 +1,255 @@
+-->
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [User Stories (Optional)](#user-stories-optional)
+    - [Story 1](#story-1)
+  - [Story 2](#story-2)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+  - [Test Plan](#test-plan)
+    - [Unit Tests](#unit-tests)
+- [Alternatives](#alternatives)
+<!-- /toc -->
+
+## Summary
+
+This KEP proposes a method for passing parameters to ProvisioningRequests. This will allow users to pass parameters specific to individual ProvisioningRequest classes.
+
+<!--
+This section is incredibly important for producing high-quality, user-focused
+documentation such as release notes or a development roadmap. It should be
+possible to collect this information before implementation begins, in order to
+avoid requiring implementors to split their attention between writing release
+notes and implementing the feature itself. KEP editors and SIG Docs
+should help to ensure that the tone and content of the `Summary` section is
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+
+Both in this section and below, follow the guidelines of the [documentation
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+
+## Motivation
+Currently, no mechanism exists to adjust these parameters for an individual Workload. Right now, users can only change the parameters with a ProvisioningRequestConfig, which then applies to all Workloads submitted to a specific ClusterQueue. This limits the usefulness/flexibility of ProvisioningRequests in Kueue.
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+### Goals
+- Provide a mechanism to pass parameters to ProvisioningRequest
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+### Non-Goals
+- Describe how the ProvisioningRequest controller handles these parameters
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+## Proposal
+We propose introducing a new group of annotations with the 
+***prov-req.kueue.x-k8s.io/ prefix***. Annotations using this prefix will be directly copied to the ProvisioningRequest's Parameters field, including both name and value.
```

Comment:
```suggestion
***alpha.prov-req.kueue.x-k8s.io/prefix***. Annotations using this prefix will be directly copied to the ProvisioningRequest's Parameters field, including both name and value.
```

@alculquicondor @PBundyra Since this feature is alpha stage, could we add a prefix, `alpha`?

Reply:
Also, I can confirm if the shorter name "prov-req" is a general term in ClusterAutoscaler repository?

Reply:
Based on [k8s well-known annotations](https://kubernetes.io/docs/reference/labels-annotations-taints/#kubeadm-alpha-kubernetes-io-cri-socket) I think it would be more aligned with convention to add `alpha` as infix,  resulting in the prefix `provreq.alpha.kueue.x-k8s.io/`. 

While `prov-req` name is indeed used in the CA repository, it's less common than `provreq`. For consistency, I propose adopting `provreq`.

Reply:
Sorry, I just got to this.
I disagree about adding `alpha` to the name of the annotation. We stopped doing this kind of thing in kubernetes, as it implies that users will need to migrate to a new annotation when this graduates.

At most, what we do in kubernetes is store the name of the label/annotation in a constant that has some form of warning that it's alpha. Like here https://github.com/kubernetes/kubernetes/blob/88e994f6bf8fc88114c5b733e09afea339bea66d/pkg/apis/batch/types.go#L445

Reply:
Oh, I see.
Thank you for pointing this out. I agree with you.
@PBundyra Could you update this KEP based on Aldo's feedbacks?

Reply:
Thanks Aldo.
I've created a [new PR](https://github.com/kubernetes-sigs/kueue/pull/1805) that corrects this.